blog/_posts/2023-05-30-new-class-tscatter.md

```diff
@@ -56,11 +53,11 @@ void scatter()
       s[i] = 400*r->Rndm(i);
    }
 
-   auto scat = new TScatter(n, x, y,c,s);
-   scat->SetMarkerStyle(20);
-   scat->SetMarkerColor(kRed);
-   scat->SetTitle("Scatter plot;X;Y");
-   scat->Draw("A");
+   auto scatter = new TScatter(n, x, y, c, s);
+   scatter->SetMarkerStyle(20);
+   scatter->SetMarkerColor(kRed);
+   scatter->SetTitle("Scatter plot;X;Y");
+   scatter->Draw("A");
```

Comment:
It would be nicer for new tutorials/code snippets to avoid bare `new` and instead use stack-based variables whenever possible, or `unique_ptr` otherwise. In this case, the following should perfectly work

```cpp
TScatter scatter{n, x, y, c, s};
scatter.SetMarkerStyle(20);
scatter.SetMarkerColor(kRed);
scatter.SetTitle("Scatter plot;X;Y");
scatter.Draw("A");
```

Reply:
Also, could you apply the same style to previous examples in this snippet? Such as `new TRandom` and `new TCanvas`.

Reply:
TScatter scatter{n, x, y, c, s};
...
scatter.Draw("A");

That does not work, it produces a blanck canvas (as expected).

Reply:
that's too bad :(

Reply:
Any chance to wrap the classes in `std::unique_ptr` at least?

Reply:
How ?

Reply:
This class follows the model of all graphics classes TGraph, TLine Txxx ... that would not make sense to code it another way.